### PR TITLE
GEN (Generalized Entropy): a new OOD detector

### PR DESCRIPTION
**Pull request by cofri**
This PR introduces a new method in OODEEL: GEN for Generalized ENtropy. It computes an entropy-like score based on softmax values. Two hyper-parameters can be set to tune the method:
- `gamma` allows to give more or less importance to minor deviations of predictions from the ideal one-hot encoded labels. The value must be between 0 and 1. The authors propose a default value of 0.1.
- `k` is the number of softmax values to keep for the entropy computation. Keeping all softmax values can cause instabilities due to very low probabilities. Keeping only the top-k values is thus recommended. For ImageNet (1000 classes), the authors set it to 100.

### CODE REVIEW

**Review by paulnovello — APPROVED** [review]
LGTM

**Review by jdalch — APPROVED** [review]
The method follows the implementation in the paper. Good job!
